dislib/data/array.py

```diff
@@ -709,7 +709,13 @@ def array(x, block_size):
         x = np.array(x, copy=True)
 
     if len(x.shape) < 2:
-        raise ValueError("Input array must have two dimensions.")
+        if block_size[0] == 1:
+            x = x.reshape(1, -1)
+        elif block_size[1] == 1:
+            x = x.reshape(-1, 1)
+        else:
+            raise ValueError("Input array is one-dimensional but "
```

Comment:
It can also be 0-dimensional (weird thing about numpy). We could also add a check that the dimension is not > 2.

Reply:
Done